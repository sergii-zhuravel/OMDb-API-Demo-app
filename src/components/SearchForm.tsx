import React, { useCallback, useEffect, useRef } from "react";
import styled from 'styled-components';


const StyledInput = styled.input`
    padding: 0.5em 1em;
    margin: 0.5em;
    color: palevioletred;
    background: papayawhip;
    border: none;
    border-radius: 15px;
    width: 50%;
`;

const Form = styled.form`
  display: flex;
  justify-content: center;
`;

export type SearchInputProps = {
    searchQuery: string;
    onSearchChange: (searchQuery: string) => void;
};

export default function SearchForm({ searchQuery, onSearchChange }: SearchInputProps) {
    const onInputChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
        onSearchChange(e.target.value);
    }, []);
    const onSubmitHandler = useCallback((e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault();
    }, []);
    const inputRef = useRef(null);
    useEffect(() => {
        if (inputRef && inputRef.current) {
            inputRef.current.focus();
        }
    }, [])
    return <Form onSubmit={onSubmitHandler}>
        <StyledInput ref={inputRef} placeholder="Enter Title" id="searchInput" type="text" name="search" aria-label={"Search Input"} onChange={onInputChange} value={searchQuery} />
    </Form>
}