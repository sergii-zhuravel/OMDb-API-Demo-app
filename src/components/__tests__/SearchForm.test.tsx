import * as React from 'react';
import { shallow } from 'enzyme';
import SearchForm from '../SearchForm';

test('SearchForm component test', () => {
    const searchQuery = "test";
    const onSearchChangeMock = jest.fn();
    const mockProps = {
        searchQuery,
        onSearchChange: onSearchChangeMock
    }
    const component = shallow(<SearchForm {...mockProps} />);

    // Snapshot 
    expect(component).toMatchSnapshot();
});