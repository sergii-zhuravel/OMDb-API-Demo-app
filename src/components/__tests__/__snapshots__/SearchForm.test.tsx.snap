// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component test 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <styled.input
    aria-label="Search Input"
    id="searchInput"
    name="search"
    onChange={[Function]}
    placeholder="Enter Title"
    type="text"
    value="test"
  />
</styled.form>
`;
